management/src/clusterm/manager/client.go

```diff
@@ -93,55 +99,67 @@ func (c *Client) doGet(rsrc string) ([]byte, error) {
 }
 
 // PostNodeCommission posts the request to commission a node
-func (c *Client) PostNodeCommission(nodeName, extraVars string) error {
-	return c.doPost(fmt.Sprintf("%s/%s", PostNodeCommissionPrefix, nodeName), extraVars, nil)
+func (c *Client) PostNodeCommission(nodeName string, flags ActionFlags) error {
```

Comment:
the client API for Commission shall take `host-group` name as separate argument instead of the struct. This keep this interface clear and explicit about the required fields as this is going to be a public facing interface for clusterm client.